### PR TITLE
block_filterd_timestamp

### PR DESCRIPTION
**Pull request by 0x070696E65**
## What is the current behavior?
Could not filter blocks by timestamp

## How have you changed the behavior?
Blocks can be filtered by timestamp

### CODE REVIEW

File: client/rest/src/db/CatapultDb.js [hunk 1]
```diff
@@ -204,11 +204,13 @@ class CatapultDb {
 	 * Retrieves filtered and paginated blocks.
 	 * @param {Uint8Array} signerPublicKey Filters by signer public key
 	 * @param {Uint8Array} beneficiaryAddress Filters by beneficiary address
+	 * @param {uint64} fromTimestamp Filters by fromTimestamp
+	 * @param {uint64} toTimestamp Filters by toTimestamp
```
Comment (Jaguar0625) [inline client/rest/src/db/CatapultDb.js:208]: try something a little more descriptive, e.g.

Filters blocks by only including blocks with a timestamp greater than or equal to provided value
Filters blocks by only including blocks with a timestamp less than or equal to provided value

File: client/rest/test/db/CatapultDb_spec.js [hunk 2]
```diff
@@ -401,7 +411,45 @@ describe('catapult db', () => {
...
-				return runTestAndVerifyIds(dbBlocks, db => db.blocks(undefined, account1.address, paginationOptions), [10]);
+				return runTestAndVerifyIds(dbBlocks, db =>
+					db.blocks(undefined, account1.address, undefined, undefined, paginationOptions), [10]);
+			});
+
+			it('fromTimestamp', () => {
+				// Arrange:
+				const dbBlocks = [
+					createBlock(10, 1, account1.publicKey, account1.address, 23456),
+					createBlock(20, 1, account1.publicKey, account1.address, 12345)
```
Comment (Jaguar0625) [inline client/rest/test/db/CatapultDb_spec.js:422]: for this test, would prefer if you add a third block with height > 23456

test is currently testing:
1. == fromTimestamp included
2. < fromTimestamp not included

but is missing
3. > fromTimestamp included

and then make similar changes to other NEW tests

File: client/rest/test/db/utils/dbTestUtils.js [hunk 3]
```diff
@@ -162,7 +162,7 @@ const createDbTransaction = (id, signerPublicKey, recipientAddress, options) =>
 		message: { size: 12, payload: new Binary(test.random.bytes(12)) },
 		mosaics: []
 	};
-	for (let j = 0; 3 < j; ++j)
+	for (let j = 0; 3 > j; ++j)
```
Comment (Jaguar0625) [inline client/rest/test/db/utils/dbTestUtils.js:165]: what is purpose of this change? 🤔

File: client/rest/src/db/CatapultDb.js (hunk 1 above)
Comment (0x070696E65) [reply client/rest/src/db/CatapultDb.js]: I used your text verbatim 🙏

File: client/rest/test/db/CatapultDb_spec.js (hunk 2 above)
Comment (0x070696E65) [reply client/rest/test/db/CatapultDb_spec.js]: I added some tests!

File: client/rest/test/db/utils/dbTestUtils.js (hunk 3 above)
Comment (0x070696E65) [reply client/rest/test/db/utils/dbTestUtils.js]: Sorry, this is a mistake. It has been restored to its original state.

**Review by Jaguar0625 — APPROVED** [review]
Nice job!

**Review by Wayonb — APPROVED** [review]

**Pull request merged**